pkg/virt-controller/watch/vmi_test.go

```diff
@@ -4206,3 +4223,20 @@ func newVMIWithGuestAgentInterface(vmi *virtv1.VirtualMachineInstance, ifaceName
 	})
 	return vmi
 }
+
+type fakeAllocator struct {
+	calls []string
+}
+
+func (alc *fakeAllocator) Sync(_ []*virtv1.VirtualMachineInstance) {
+	alc.calls = append(alc.calls, "Sync")
+}
+
+func (alc *fakeAllocator) Allocate(_ *virtv1.VirtualMachineInstance) error {
+	alc.calls = append(alc.calls, "Allocate")
+	return nil
+}
+
+func (alc *fakeAllocator) Remove(key string) {
+	alc.calls = append(alc.calls, "Remove")
+}
```

Comment:
How about create a pkg `fake` and put this in a `vsock.go` file?
Other fake stuff can be put there.
Generally I think it would be great to have always this pattern

Reply:
After the vsock pkg creation it will be moved under `/vsock/fake/allocator` maybe

Reply:
I think this might be too simple and I don't expect anyone reusing this fake.